File: android/app/src/main/java/com/freelancekar/app/ui/screens/PaymentsScreen.kt
package com.freelancekar.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.freelancekar.app.data.*
import com.freelancekar.app.ui.AppViewModel
import com.freelancekar.app.ui.components.BackButton
import com.freelancekar.app.ui.components.InlineButton
import com.freelancekar.app.ui.components.InlineOutlineButton
import com.freelancekar.app.ui.components.SectionTitle
import com.freelancekar.app.ui.theme.FkColor
import com.freelancekar.app.ui.theme.FkType

@Composable
fun PaymentsScreen(state: AppState, vm: AppViewModel) {
    val totals = paymentTotals(state)

    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState())) {
        Row(Modifier.fillMaxWidth().padding(horizontal = 20.dp).padding(top = 16.dp), verticalAlignment = Alignment.CenterVertically) {
            BackButton { vm.go(Screen.FDASH) }
            Spacer(Modifier.width(8.dp))
            Text("Payments", style = FkType.screenTitle)
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 18.dp)) {
            Column(
                Modifier.fillMaxWidth().clip(RoundedCornerShape(20.dp)).background(FkColor.lime).padding(18.dp),
            ) {
                Text("COLLECTED · AUGUST", style = FkType.mono.copy(color = FkColor.onLime.copy(alpha = 0.65f)))
                Spacer(Modifier.height(8.dp))
                Text(money(totals.payGot), style = FkType.big.copy(color = FkColor.onLime))
                Row(
                    Modifier.fillMaxWidth().padding(top = 12.dp).border(0.dp, FkColor.onLimeDivider).padding(top = 12.dp),
                    horizontalArrangement = Arrangement.spacedBy(18.dp),
                ) {
                    PayFigure(money(totals.payDue), "still to collect")
                    PayFigure(money(totals.payTotal), "billed this month")
                    PayFigure("₹0", "platform cut")
                }
            }
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 22.dp)) {
            SectionTitle("To collect", right = { Text("PAID TO YOU DIRECTLY", style = FkType.monoSm.copy(color = FkColor.text40)) })
            Spacer(Modifier.height(11.dp))
            if (totals.due.isEmpty()) {
                Box(
                    Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(15.dp))
                        .background(FkColor.card)
                        .border(1.dp, FkColor.line14, RoundedCornerShape(15.dp))
                        .padding(20.dp),
                ) {
                    Text(
                        "Nothing outstanding. Every client has paid up.",
                        style = FkType.bodySm.copy(color = FkColor.text45),
                        modifier = Modifier.fillMaxWidth(),
                        textAlign = TextAlign.Center,
                    )
                }
            } else {
                Column(verticalArrangement = Arrangement.spacedBy(9.dp)) {
                    totals.due.forEach { p ->
                        Column(
                            Modifier
                                .fillMaxWidth()
                                .clip(RoundedCornerShape(15.dp))
                                .background(FkColor.card)
                                .border(1.dp, FkColor.line08, RoundedCornerShape(15.dp))
                                .padding(horizontal = 14.dp, vertical = 13.dp),
                        ) {
                            Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                                Text(p.client, style = FkType.cardTitle)
                                Text(money(p.amount), style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                            }
                            Spacer(Modifier.height(4.dp))
                            Text("${p.service} · ${p.whenText}", style = FkType.meta.copy(color = FkColor.text50))
                            Spacer(Modifier.height(11.dp))
                            Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                                InlineOutlineButton({ vm.remindPayment(p.client) }, text = "Remind", modifier = Modifier.weight(1f))
                                InlineButton({ vm.markPaymentReceived(p.id) }, text = "Mark received", modifier = Modifier.weight(1.3f))
                            }
                        }
                    }
                }
            }
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 22.dp)) {
            Text("Received", style = FkType.h3)
            Spacer(Modifier.height(11.dp))
            Column(
                Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(15.dp))
                    .background(FkColor.card)
                    .border(1.dp, FkColor.line08, RoundedCornerShape(15.dp)),
            ) {
                totals.got.forEach { p ->
                    Row(
                        Modifier.fillMaxWidth().border(0.dp, FkColor.line07).padding(horizontal = 14.dp, vertical = 13.dp),
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
                            Text(p.client, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                            Text("${p.service} · ${p.via}", style = FkType.meta.copy(color = FkColor.text50))
                        }
                        Text(money(p.amount), style = FkType.bodySm.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                    }
                }
            }
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 22.dp, bottom = 26.dp)) {
            Text("Your plan", style = FkType.h3)
            Spacer(Modifier.height(11.dp))
            Column(
                Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(15.dp))
                    .background(FkColor.card)
                    .border(1.dp, FkColor.line08, RoundedCornerShape(15.dp))
                    .padding(14.dp),
            ) {
                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                    Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
                        Text("${planName(state)} · ${planPrice(state)}", style = FkType.cardTitle)
                        Text("Renews 1 Sep · Razorpay", style = FkType.meta.copy(color = FkColor.text50))
                    }
                    Box(
                        Modifier
                            .clip(RoundedCornerShape(10.dp))
                            .border(1.dp, FkColor.line14, RoundedCornerShape(10.dp))
                            .clickable { vm.managePlan() }
                            .padding(horizontal = 13.dp, vertical = 9.dp),
                    ) {
                        Text("Manage", style = FkType.meta.copy(color = FkColor.text.copy(alpha = 0.75f), fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                    }
                }
                Row(
                    Modifier.fillMaxWidth().padding(top = 12.dp).border(0.dp, FkColor.line07).padding(top = 12.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
                        Text("Where clients pay you", style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                        Text(if (state.bank) "HDFC ••4471 · verified" else "Add UPI ID or account number", style = FkType.meta.copy(color = FkColor.text50))
                    }
                    Box(
                        Modifier
                            .clip(RoundedCornerShape(10.dp))
                            .background(if (state.bank) FkColor.limeFill else FkColor.lime)
                            .clickable { vm.toggleBank() }
                            .padding(horizontal = 13.dp, vertical = 9.dp),
                    ) {
                        Text(if (state.bank) "Added" else "Add", style = FkType.meta.copy(color = if (state.bank) FkColor.lime else FkColor.onLime, fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                    }
                }
            }
            Spacer(Modifier.height(13.dp))
            Text(
                "FreelanceKar never holds your money — clients pay you by UPI or cash and you keep every rupee.",
                style = FkType.meta.copy(color = FkColor.text40),
            )
        }
    }
}

@Composable
private fun PayFigure(value: String, label: String) {
    Column {
        Text(value, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold, fontSize = androidx.compose.ui.unit.TextUnit(14f, androidx.compose.ui.unit.TextUnitType.Sp), color = FkColor.onLime))
        Text(label, style = FkType.metaXs.copy(color = FkColor.onLime.copy(alpha = 0.65f)))
    }
}

File: ios/FreelanceKar/Views/Screens/PaymentsScreen.swift
import SwiftUI

struct PaymentsScreen: View {
    @EnvironmentObject var store: AppStore

    var body: some View {
        let s = store.state
        let totals = Selectors.paymentTotals(s)

        ScreenShell {
            VStack(alignment: .leading, spacing: 0) {
                HStack(spacing: 12) {
                    BackButton { store.go(.fdash) }
                    Text("Payments").font(.fkScreenTitle)
                }
                .padding(.horizontal, 20).padding(.top, 16)

                VStack(alignment: .leading, spacing: 0) {
                    Text("COLLECTED · AUGUST").font(.system(size: 11, weight: .medium, design: .monospaced)).opacity(0.65)
                    Text(money(totals.payGot)).font(.fkBig).padding(.top, 8)
                    HStack(spacing: 18) {
                        Figure(value: money(totals.payDue), label: "still to collect")
                        Figure(value: money(totals.payTotal), label: "billed this month")
                        Figure(value: "₹0", label: "platform cut")
                    }
                    .padding(.top, 12)
                    .overlay(Rectangle().frame(height: 1).foregroundStyle(FK.onLimeDivider), alignment: .top)
                    .padding(.top, 12)
                }
                .padding(18)
                .foregroundStyle(FK.onLime)
                .background(FK.lime)
                .clipShape(RoundedRectangle(cornerRadius: 20, style: .continuous))
                .padding(.horizontal, 20).padding(.top, 18)

                VStack(alignment: .leading, spacing: 11) {
                    SectionTitle(title: "To collect") {
                        Text("PAID TO YOU DIRECTLY").font(.fkMonoSm).foregroundStyle(FK.text40)
                    }
                    if totals.due.isEmpty {
                        Text("Nothing outstanding. Every client has paid up.")
                            .font(.fkBodySm).foregroundStyle(FK.text45).multilineTextAlignment(.center)
                            .frame(maxWidth: .infinity).padding(20)
                            .background(FK.card)
                            .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
                            .overlay(RoundedRectangle(cornerRadius: 15).stroke(style: StrokeStyle(lineWidth: 1, dash: [5, 4])).foregroundStyle(FK.line14))
                    } else {
                        ForEach(totals.due) { p in
                            VStack(alignment: .leading, spacing: 0) {
                                HStack {
                                    Text(p.client).font(.fkCardTitle)
                                    Spacer()
                                    Text(money(p.amount)).font(.system(size: 13, weight: .semibold))
                                }
                                Text("\(p.service) · \(p.when)").font(.system(size: 11.5)).foregroundStyle(FK.text50).padding(.top, 4)
                                HStack(spacing: 8) {
                                    InlineOutlineButton(title: "Remind", expand: true) { store.remindPayment(p.client) }
                                    InlineButton(title: "Mark received", expand: true) { store.markPaymentReceived(p.id) }
                                }.padding(.top, 11)
                            }
                            .padding(14)
                            .background(FK.card)
                            .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
                            .overlay(RoundedRectangle(cornerRadius: 15).stroke(FK.line08, lineWidth: 1))
                        }
                    }
                }
                .padding(.horizontal, 20).padding(.top, 22)

                VStack(alignment: .leading, spacing: 11) {
                    Text("Received").font(.fkH3)
                    VStack(spacing: 0) {
                        ForEach(Array(totals.got.enumerated()), id: \.element.id) { i, p in
                            HStack(spacing: 11) {
                                VStack(alignment: .leading, spacing: 2) {
                                    Text(p.client).font(.system(size: 13, weight: .semibold))
                                    Text("\(p.service) · \(p.via)").font(.system(size: 11)).foregroundStyle(FK.text50)
                                }
                                Spacer()
                                Text(money(p.amount)).font(.system(size: 12.5, weight: .semibold))
                            }
                            .padding(.horizontal, 14).padding(.vertical, 13)
                            .overlay(i < totals.got.count - 1 ? Rectangle().frame(height: 1).foregroundStyle(FK.line07) : nil, alignment: .bottom)
                        }
                    }
                    .background(FK.card)
                    .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
                    .overlay(RoundedRectangle(cornerRadius: 15).stroke(FK.line08, lineWidth: 1))
                }
                .padding(.horizontal, 20).padding(.top, 22)

                VStack(alignment: .leading, spacing: 11) {
                    Text("Your plan").font(.fkH3)
                    VStack(spacing: 0) {
                        HStack {
                            VStack(alignment: .leading, spacing: 3) {
                                Text("\(Selectors.planName(s)) · \(Selectors.planPrice(s))").font(.fkCardTitle)
                                Text("Renews 1 Sep · Razorpay").font(.system(size: 11)).foregroundStyle(FK.text50)
                            }
                            Spacer()
                            Button { store.managePlan() } label: {
                                Text("Manage").font(.system(size: 11.5, weight: .semibold)).foregroundStyle(Color(hex: 0xF2F2F0, opacity: 0.75))
                                    .padding(.horizontal, 13).padding(.vertical, 9)
                            }
                            .overlay(RoundedRectangle(cornerRadius: 10).stroke(FK.line14, lineWidth: 1))
                        }
                        HStack {
                            VStack(alignment: .leading, spacing: 3) {
                                Text("Where clients pay you").font(.system(size: 13, weight: .semibold))
                                Text(s.bank ? "HDFC ••4471 · verified" : "Add UPI ID or account number").font(.system(size: 11)).foregroundStyle(FK.text50)
                            }
                            Spacer()
                            Button { store.toggleBank() } label: {
                                Text(s.bank ? "Added" : "Add").font(.system(size: 11.5, weight: .semibold))
                                    .foregroundStyle(s.bank ? FK.lime : FK.onLime)
                                    .padding(.horizontal, 13).padding(.vertical, 9)
                            }
                            .background(s.bank ? FK.limeFill : FK.lime)
                            .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
                        }
                        .padding(.top, 12)
                        .overlay(Rectangle().frame(height: 1).foregroundStyle(FK.line07), alignment: .top)
                        .padding(.top, 12)
                    }
                    .padding(14)
                    .background(FK.card)
                    .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
                    .overlay(RoundedRectangle(cornerRadius: 15).stroke(FK.line08, lineWidth: 1))

                    Text("FreelanceKar never holds your money — clients pay you by UPI or cash and you keep every rupee.")
                        .font(.system(size: 11.5)).foregroundStyle(FK.text40).lineSpacing(3).padding(.top, 13)
                }
                .padding(.horizontal, 20).padding(.top, 22).padding(.bottom, 26)
            }
        }
    }
}

private struct Figure: View {
    var value: String
    var label: String
    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            Text(value).font(.system(size: 14, weight: .semibold))
            Text(label).font(.system(size: 10.5)).opacity(0.65)
        }
    }
}
